src/components/hmi_message_handler/test/hmi_message_handler_impl_test.cc

```diff
@@ -189,6 +189,46 @@ TEST_F(HMIMessageHandlerImplTest, SendMessageToHMI_Success) {
   EXPECT_TRUE(waiter.WaitFor(1, 100));
 }
 
+TEST(WebsocketSessionTest, SendMessage_UnpreparedConnection_WithoutFall) {
+  ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+
+  auto send_message = []() {
+    auto message =
+        "{\"id\" : 1,\"jsonrpc\" : \"2.0\",\"method\" : "
+        "\"BasicCommunication.GetSystemInfo\"}";
+
+    Json::Reader reader;
+    Json::Value json_value;
+
+    ASSERT_TRUE(reader.parse(message, json_value, false));
+
+    // Make unprepared connection
+    boost::asio::io_context ioc{1};
+    boost::asio::ip::tcp::acceptor acceptor{
+        ioc, {boost::asio::ip::make_address("127.0.0.1"), 8087}};
+    boost::asio::ip::tcp::socket socket{ioc};
+
+    std::unique_ptr<hmi_message_handler::WebsocketSession> session(
+        new hmi_message_handler::WebsocketSession(std::move(socket), nullptr));
+
+    // Send message to unprepared connection
+    session->sendJsonMessage(json_value);
+
+    // Wait for the message to be processed
```

Comment:
DO we have an ability to catch processed event? 
I definitely don't like adding sleep in tests.

Reply:
@LuxoftAKutsan 
In this case don't.
I see only one method to avoid sleep.
We can inherit class in unit test from WebsocketSession but also we need change current implementation for the WebsocketSession, need change private section to protected for accessing to message_queue_.

Reply:
Ok, I see, that issue is that LoopThreadDelegate requires WebsocketSession to work. 
WebsocketSession  init LoopThreadDelegate  with `this`, so we have no ability to mock it without additional setters. 

But LoopThreadDelegate  uses only socket from WebsocketSession, that you pass to WebsocketSession  constructor. So LoopThreadDelegate does not need WebsocketSession, only socket. 

Good solution would be creating wrapper class for socket, then providing mock of the socket to WebsocketSession in unit tests. 
You will ne to wrap following interfaces : 
 - async_read 
 - write
 - get_executor
 - async_accept

I feel it like to big changes for this PR.  Please create enhancement ticket and following comment in this unit test. Comment should contain why sleep is used, and what refactoring should be applied to avoid sleeps